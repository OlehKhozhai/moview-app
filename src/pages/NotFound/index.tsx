import React from 'react';

import Button from 'components/_common/Button';
import Logo from 'components/_common/Logo';
import styles from './styles.module.scss';

const NotFound = () => {
  return (
    <div className={styles.root}>
      <Logo className={styles.logo} />
      <h2 className={styles.title}>Page not found</h2>
      <Button title="Go back to home" linkTo="/" variation="primaryOutline" />
    </div>
  );
};

export default React.memo(NotFound);
