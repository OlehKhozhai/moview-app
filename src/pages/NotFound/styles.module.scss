@import 'assets/styles/mixins/colors';

.root {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: column;
  position: relative;
  height: 100vh;
}

.logo {
  position: absolute;
  top: 30px;
  left: 0;
}

.title {
  font-size: 50px;
  color: $primary-color;
  margin-bottom: 40px;
}
