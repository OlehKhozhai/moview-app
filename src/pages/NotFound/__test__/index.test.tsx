import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import NotFound from '../index';

describe('NotFound component', () => {
  test('should render markup by default', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <NotFound />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
