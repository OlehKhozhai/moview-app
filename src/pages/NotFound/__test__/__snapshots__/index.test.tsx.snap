// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render markup by default 1`] = `
<div
  className="root"
>
  <a
    className="root logo"
    href="/"
    onClick={[Function]}
  >
    <span
      className="bold"
    >
      netflix
    </span>
    roulete
  </a>
  <h2
    className="title"
  >
    Page not found
  </h2>
  <a
    className="root primaryOutline"
    href="/"
    onClick={[Function]}
  >
    Go back to home
  </a>
</div>
`;
